ArduCopter/events.cpp

```diff
@@ -355,7 +360,8 @@ bool Copter::should_disarm_on_failsafe() {
             return !ap.auto_armed && ap.land_complete;
         case Mode::Number::GUIDED:
             if (g.failsafe_throttle == FS_THR_ENABLED_CONTINUE_MISSION ||
-                g.failsafe_throttle == FS_THR_ENABLED_CONTINUE_MISSION_ALWAYS_LAND) {
+                g.failsafe_throttle == FS_THR_ENABLED_CONTINUE_MISSION_ALWAYS_LAND ||
+                g.failsafe_throttle == FS_THR_ENABLED_ALWAYS_LAND) {
```

Comment:
this should check failsafe_option(FailsafeOption::RC_CONTINUE_IF_AUTO) as well

Reply:
Okay added, also added 'RC_CONTINUE_IF_AUTO', as we would need the same behavior for Guided

Reply:
Why should this check `RC_CONTINUE_IF_AUTO` if the case entry requires that the aircraft is in GUIDED? Should this bit check be in the `AUTO` case?

Reply:
It should be for both the modes, AUTO and GUIDED. So added it for AUTO as well as GUIDED